src/main/java/org/spongepowered/common/event/SpongeEventManager.java

```diff
@@ -239,7 +249,7 @@ private PluginContainer getPlugin(Object plugin) {
 
     @Override
     public void registerListeners(Object plugin, Object listener) {
-        registerListener(getPlugin(plugin), listener);
+        registerListeners(getPlugin(plugin), listener);
```

Comment:
How does this not cause a stack overflow?

Reply:
@gabizou in the reincarnation over on https://github.com/SpongePowered/SpongeCommon/pull/1737/ I've undone this part, if that helps to finally get this in? That also takes care of any possible backwards compatibility concerns.